pkg/component/worker/kubelet.go

```diff
@@ -170,15 +192,25 @@ func (k *Kubelet) Start(ctx context.Context) error {
 		args["--node-labels"] = strings.Join(k.Labels, ",")
 	}
 
-	if runtime.GOOS == "windows" {
-		node, err := node.GetNodename("")
+	nodename, err := node.GetNodename("")
```

Comment:
We need to honor the `--hostname-override` flag here. Maybe move up the extra args splitting here and pass the override flag value instead of the empty string?